pages/data-modeling/access-policies.mdx

```diff
@@ -0,0 +1,450 @@
+- You can define data access policies in both cubes and views. We recommend [keeping cubes private](/data-modeling/views#how-to-define-cube-views) and defining access policies only in views. This makes policies easier to maintain and reason about, especially for member-level access.
+
+**Deny all by default**
+
+Access policies already restrict access by default i.e. if a user’s role does not match any policy, they will not see any data.
+
+Sometimes you may still want to make this explicit in your model. You can do this by setting `includes: []` for the `*` role. This doesn’t change behavior, but it documents the intent clearly and avoids assumptions.
+
+```yaml
+access_policy:
+  - role: "*"
+    member_level:
+      includes: []
+
+  - role: employee
+    member_level:
+      excludes:
+        - revenue
+        - profit_margin
+    row_level:
+      filters:
+        - member: department
+          operator: equals
+          values: ["{ securityContext.department }"]
+
+  - role: manager
+    member_level:
+      includes: "*"
+    row_level:
+      filters:
+        - member: department
+          operator: equals
+          values: ["{ securityContext.department }"]
+```
+
+**Things to notice:**
+
+- The  role with `includes: []` removes access for all users by default.
+- Only roles explicitly defined (`employee`, `manager`) are granted access.
+- The `manager` role also filters rows by department, but includes all fields in the model, including revenue-related measures.
+- This approach follows the principle of least privilege: start with no access, then grant access selectively.
+
+## Testing Access Policies
+
+Switch between different security contexts in the no-code builder to test how roles affect visibility.
+
+**Example 1: Admin**
+
+```yaml
+# Admin sees all rows and all fields
+- name: Admin
+  securityContext:
+    user_id: 0
+  roles:
+    - admin
+```
+
+**Example 2: User**
+
+```yaml
+- name: User
+  securityContext:
+    user_id: 123
+  roles:
+    - user
+```
+
+**Customers Model:**
+
+```yaml
+access_policy:
+  - role: admin
+    member_level:
+      includes: "*" 
+    row_level:
+      allow_all: true
+
+  - role: user
+    member_level:
+      includes: "*" 
+    row_level:
+      filters:
+        - member: id
+          operator: equals
+          values: ["{ securityContext.user_id }"]
+```
+
+<VideoComponent
+    src="/video/access_policies_example.mp4"
+    width="1920"
+    height="1080"
+/>
+
+## Examples
+
+### Multiple values
+
+This example shows how to allow access to multiple regions at once by passing them in the token.
+
+**Token**
+
+```yaml
+- name: Default
+  securityContext:
+    regions: [us-east, eu-west]
+  roles:
+    - default
+```
+
+**Access policy**
+
+```yaml
+access_policy:
+  - role: default
+    row_level:
+      filters:
+        - member: region
+          operator: equals
+          values: ["{ securityContext.regions }"]
```

Comment:
@Nauman1011 are you sure this works?

I was under the impression one has to do something like this:

```
          values: { [securityContext.regions] }
```